lib/versioning/regex/index.ts

```diff
@@ -0,0 +1,125 @@
+import { compare, satisfies, ltr, minSatisfying, maxSatisfying } from 'semver';
+import { VersioningApiConstructor } from '../common';
+import { GenericVersion, GenericVersioningApi } from '../loose/generic';
+
+export interface RegExpVersion extends GenericVersion {
+  // prereleases are treated in the standard semver manner, if present
+  prerelease: string;
+  // architectures, if present, are treated as a compatibility layer: we will
+  // never try to update to a version with a different architecture. Other than
+  // for juding compatibility (exact string match), the architecture is ignored
+  architecture: string;
```

Comment:
I'm still thinking whether we should simply call this "compatibility"?

Reply:
we can do that, we can also reuse `suffix` from base class `GenericVersion`

Reply:
`suffix` is a bit too specific though. can you refactor it to `compatibility`?

Reply:
ok, will also just replace the comment? or should we leave them as is?

Reply:
yeah, the comment can be simplified then